packages/lib/src/FormControllerClass/FormControllerClass.ts

```diff
@@ -522,24 +522,52 @@ export class FormControllerClass {
     await this.validate();
 
     const [errors, values] = toJSCompat([this.API.errors, this.API.values]);
-    const isSuccess = isEmpty(errors);
+    const isValid = isEmpty(errors);
+
+    let isSuccess: boolean = false;
+    let error: unknown;
+    let response: unknown;
 
     try {
       if (this.options.onSubmit) {
-        await this.options.onSubmit({errors, values, isSuccess, event: submitEvent});
+        response = await this.options.onSubmit({
+          errors,
+          values,
+          isValid,
+          isSuccess: isValid,
+          event: submitEvent,
+        });
       }
 
+      isSuccess = true;
+
       runInAction(() => {
         if (isEmpty(errors)) {
           this.updateInitialValues();
           this.updateIsDirtyBasedOnFields();
         }
         this.setIsSubmitting(false);
       });
-    } catch {
+    } catch (e) {
+      error = e;
+
       this.setIsSubmitting(false);
     }
 
-    return {errors, values, isSuccess};
+    return {
+      errors,
+      values,
+      isValid,
+      isSuccess: isValid,
```

Comment:
isSuccess,

Reply:
I left isSuccess here for compatibility reasons so we could migrate to isValid in our projects. But as I mentioned before - name is confusing and it should have been isValid. So I keep both until we migrate, then remove isSuccess from here.